src/EditorFeatures/Core/Classification/TotalClassificationTaggerProvider.cs

```diff
@@ -0,0 +1,216 @@
+    private readonly SemanticClassificationViewTaggerProvider _semanticTaggerProvider = new(threadingContext, typeMap, globalOptions, visibilityTracker, listenerProvider);
+    private readonly EmbeddedLanguageClassificationViewTaggerProvider _embeddedTaggerProvider = new(threadingContext, typeMap, globalOptions, visibilityTracker, listenerProvider);
+
+    public ITagger<T>? CreateTagger<T>(ITextView textView, ITextBuffer buffer) where T : ITag
+    {
+        var syntacticTagger = _syntacticTaggerProvider.CreateTagger(buffer);
+        var semanticTagger = _semanticTaggerProvider.CreateTagger(textView, buffer);
+        var embeddedTagger = _embeddedTaggerProvider.CreateTagger(textView, buffer);
+
+        if (syntacticTagger is null || semanticTagger is null || embeddedTagger is null)
+        {
+            syntacticTagger?.Dispose();
+            semanticTagger?.Dispose();
+            embeddedTagger?.Dispose();
+            return null;
+        }
+
+        var finalTagger = new TotalClassificationAggregateTagger(syntacticTagger, semanticTagger, embeddedTagger);
+        if (finalTagger is not ITagger<T> typedTagger)
+        {
+            finalTagger.Dispose();
+            return null;
+        }
+
+        return typedTagger;
+    }
+
+    private sealed class TotalClassificationAggregateTagger(
+        SyntacticClassificationTaggerProvider.Tagger syntacticTagger,
+        SimpleAggregateTagger<IClassificationTag> semanticTagger,
+        SimpleAggregateTagger<IClassificationTag> embeddedTagger)
+        : AbstractAggregateTagger<IClassificationTag>(ImmutableArray.Create<ITagger<IClassificationTag>>(syntacticTagger, semanticTagger, embeddedTagger))
+    {
+        private static readonly Comparison<ITagSpan<IClassificationTag>> s_spanComparison = static (s1, s2) => s1.Span.Start - s2.Span.Start;
+
+        public override SegmentedList<ITagSpan<IClassificationTag>> GetTags(NormalizedSnapshotSpanCollection spans)
+        {
+            // First, get all the syntactic tags.  While they are generally overridden by semantic tags (since semantics
+            // allows us to understand better what things like identifiers mean), they do take precedence for certain
+            // tags like 'Comments' and 'Excluded Code'.  In those cases we want the classification to 'snap' instantly to
+            // the syntactic state, and we do not want things like semantic classifications showing up over that.
+
+            var totalTags = new SegmentedList<ITagSpan<IClassificationTag>>();
+
+            using var _ = Classifier.GetPooledList<ITagSpan<IClassificationTag>>(out var stringLiterals);
+
+            var syntacticSpans = syntacticTagger.GetTags(spans).GetEnumerator();
+            var semanticSpans = semanticTagger.GetTags(spans).GetEnumerator();
+
+            var currentSyntactic = NextOrNull(ref syntacticSpans);
+            var currentSemantic = NextOrNull(ref semanticSpans);
```

Comment:
this follows a similar pattern we use when we produce tags and diff them against the last produced tags.